src/App.tsx

```diff
@@ -1,61 +1,119 @@
 import './App.scss';
+import { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './components/TodoInfo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+export const findUser = (userId: number) => {
+  return usersFromServer.find(user => user.id === userId) || null;
+};
+
+export const todosData: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: findUser(todo.userId),
+}));
 
 export const App = () => {
+  const [usertitle, setUserTitle] = useState('');
+  const [errorTitle, setErrorTitle] = useState(false);
+  const [errorName, setErrorName] = useState(false);
```

Comment:
it's not a clear name for the boolean variable, you can rename these to [ hasErrorTitle, setHasErrorTitle ] or 
[ isErrorTitle, setIsErrorTitle ]

Reply:
still not resolved